src/coreclr/crossgen-corelib.sh

```diff
@@ -94,7 +94,7 @@ __UnprocessedBuildArgs=
 
 source "$__ProjectRoot"/_build-commons.sh
 
-if [[ "${__BuildArch}" != "${__HostArch}" ]]; then
+if [[ "${__BuildArch}" != "${__HostArch}" ]] || [[ "$__BuildOS" != "$__HostOS" ]]; then
```

Comment:
Is anything setting `__BuildOS`?

Reply:
I missed file in the update. I updated build-commons.sh to have three flavors you mentioned, all defaulted to current os.